File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.py
import sys
input = sys.stdin.readline


N, M = map(int, input().split())

my_dict = dict()

for i in range(1, N+1):
    name = input().strip()
    # 이름과 번호를 하나의 dict에서 다 관리
    my_dict[str(i)] = name
    my_dict[name] = str(i)

print('\n'.join([my_dict[input().strip()] for _ in range(M)]))

File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.kt
package org.example


fun main() {
    val br = System.`in`.bufferedReader()
    val sb = StringBuilder()
    val (N, M) = br.readLine().split(" ").map { it.toInt() }
    // 포켓몬 수 n 맞출 m
    val numberName = mutableListOf<String>()
    val nameHashMap = HashMap<String, Int>()
    for (number in 1..N) {
        // 포켓몬 1번 ~ N번까지 들어옴
        val name = br.readLine()
        numberName.add(name)
        nameHashMap[name] = number
    }
    repeat(M) {
        val userInput = br.readLine()
        if (userInput[0].isDigit()) { // 첫번째 글자만 숫자인지 판단해줘도 됨
            sb.append(numberName[userInput.toInt() - 1]).append("\n")
        } else {
            sb.append(nameHashMap.getValue(userInput)).append("\n")
        }
    }
    print(sb)
    br.close()
}